tools/slicec-cs/src/generators/enum_generator.rs

```diff
@@ -11,35 +13,102 @@ use slicec::grammar::*;
 pub fn generate_enum(enum_def: &Enum) -> CodeBlock {
     let mut code = CodeBlock::default();
     code.add_block(&enum_declaration(enum_def));
-    code.add_block(&enum_underlying_extensions(enum_def));
+
+    if enum_def.is_mapped_to_cs_enum() {
+        code.add_block(&enum_underlying_extensions(enum_def));
+    }
+
     code.add_block(&enum_encoder_extensions(enum_def));
     code.add_block(&enum_decoder_extensions(enum_def));
     code
 }
 
 fn enum_declaration(enum_def: &Enum) -> CodeBlock {
-    let mut builder = ContainerBuilder::new(
-        &format!("{} enum", enum_def.access_modifier()),
-        &enum_def.escape_identifier(),
-    );
-    if let Some(summary) = enum_def.formatted_doc_comment_summary() {
-        builder.add_comment("summary", summary);
-    }
-    builder
-        .add_generated_remark("enum", enum_def)
-        .add_comments(enum_def.formatted_doc_comment_seealso())
-        .add_obsolete_attribute(enum_def)
-        .add_base(enum_def.get_underlying_cs_type())
-        .add_block(enum_values(enum_def));
-
-    // Add cs::attribute
-    for attribute in enum_def.cs_attributes() {
-        builder.add_attribute(attribute);
+    if enum_def.is_mapped_to_cs_enum() {
+        // Mapped to a C# enum.
+        let mut builder = ContainerBuilder::new(
+            &format!("{} enum", enum_def.access_modifier()),
+            &enum_def.escape_identifier(),
+        );
+        if let Some(summary) = enum_def.formatted_doc_comment_summary() {
+            builder.add_comment("summary", summary);
+        }
+        builder
+            .add_generated_remark("enum", enum_def)
+            .add_comments(enum_def.formatted_doc_comment_seealso())
+            .add_obsolete_attribute(enum_def)
+            .add_base(enum_def.get_underlying_cs_type())
+            .add_block(enumerators(enum_def));
+
+        // Add cs::attribute
+        for attribute in enum_def.cs_attributes() {
+            builder.add_attribute(attribute);
+        }
+        builder.build()
+    } else {
+        // Mapped to a Dunet discriminated union.
+
+        let mut builder = ContainerBuilder::new(
+            &format!(
+                r#"
+[Dunet.Union]
+{} abstract partial record class"#,
+                enum_def.access_modifier()
```

Comment:
Should be a trailing comma here
```suggestion
                enum_def.access_modifier(),
```